xen/drivers/passthrough/arm/io-pgtable-arm.c

```diff
@@ -326,21 +418,32 @@ static int __arm_lpae_map(struct arm_lpae_io_pgtable *data, unsigned long iova,
 	/* Grab a pointer to the next level */
 	pte = *ptep;
 	if (!pte) {
-		cptep = __arm_lpae_alloc_pages(ARM_LPAE_GRANULE(data),
+		page = __arm_lpae_alloc_pages(ARM_LPAE_GRANULE(data),
 					       GFP_ATOMIC, cfg);
-		if (!cptep)
+		if (!page)
```

Comment:
This __arm_lpae_alloc_pages is called three times and I check for (!...) everywhere.